andela/caret-bn-backend
# Ft integrate requests trips 168781688

#### What does this PR do?

Allows a user to view requests that they have made on the barefoot nomad platform from over a period of time.

#### Description of Task to be completed?

- Allow a user to view their requests by hitting an endpoint.
- Create relations between the requests table, the locations table, the destinations & the request statuses.

#### How should this be manually tested?

1. Clone this repo
2. Run `npm install `to install all dependencies
3. Run ` sequelize db:migrate ` to migrate all the tables
4. Run ` sequelize db:seed:all ` to seed the default values for locations, requestStatus  & tripTypes tables.  
5. Sign up to the platform / login - take note of the id & token properties in the data object of the response
6. Manually create a record in the requests table. (Default accomodation to the value 1). Refer to the seeded location table for locationId value and the seeded tripTypes table for typeId value.
7. Manually create a record in the destination table. Refer to seeded tables for the corresponding Id values. 
8. With the token in the header [bearer token], use postman to make a ` GET ` request to ` localhost:3000/api/v1/requests ` 
9. Verify that the correct data is returned.

#### Any background context you want to provide?

N / A

#### What are the relevant pivotal tracker stories?

###### #168781688

#### Screenshots (if appropriate)

<img width="1341" alt="Screen Shot 2019-10-16 at 11 04 23" src="https://user-images.githubusercontent.com/13886438/66904738-e7bd7780-f004-11e9-871b-542cb39f9f81.png">
<img width="1375" alt="Screen Shot 2019-10-16 at 11 04 30" src="https://user-images.githubusercontent.com/13886438/66904740-e8560e00-f004-11e9-9dfa-8938d94a08c1.png">
<img width="814" alt="Screen Shot 2019-10-16 at 11 04 46" src="https://user-images.githubusercontent.com/13886438/66904741-e8560e00-f004-11e9-9a70-ff12f9e396fb.png">
<img width="1275" alt="Screen Shot 2019-10-16 at 11 05 01" src="https://user-images.githubusercontent.com/13886438/66904743-e8560e00-f004-11e9-8a9b-c3a52a8de5be.png">


#### Questions:
N / A
